pkgs/development/compilers/tinygo/default.nix

```
@@ -1,64 +1,129 @@
-{ lib, buildGoModule, fetchFromGitHub, llvm, clang-unwrapped, lld, avrgcc
-, avrdude, openocd, gcc-arm-embedded, makeWrapper, fetchurl }:
+{ stdenv
+, lib
+, buildGoModule
+, fetchFromGitHub
+, makeWrapper
+, substituteAll
+, llvmPackages
+, libffi
+, wasi-libc
+, avrgcc
+, go
+, binaryen
+, avrdude
+, gdb
+, openocd
+}:
 
-let main = ./main.go;
-    gomod = ./go.mod;
+let
+  llvmMajor = lib.versions.major llvm.version;
+  inherit (llvmPackages) llvm clang compiler-rt lld;
 in
+
 buildGoModule rec {
   pname = "tinygo";
-  version = "0.16.0";
+  version = "0.23.0";
 
   src = fetchFromGitHub {
     owner = "tinygo-org";
-    repo = "tinygo";
+    repo = pname;
```

```suggestion
    repo = "tinygo";
```
I like this way much more because it is very obvious from which source things are comming.